spectacles/validators/content.py

```diff
@@ -22,6 +31,32 @@ def _get_personal_folders(self) -> List[int]:
                 personal_folders.append(folder["id"])
         return personal_folders
 
+    def _get_all_subfolders(self, folders: List[int]) -> List[int]:
+        result = []
+        for folder in folders:
+            result.extend(self._get_subfolders(folder))
+        return result
+
+    def _get_subfolders(self, folder_id: int) -> List[int]:
+        all_folders = self.client.all_folders(self.project.name)
```

Comment:
Instead of getting all folders every time (which could get hefty), let's use the [Search Folder endpoint](https://docs.looker.com/reference/api-and-integration/api-reference/v3.1/folder#search_folders) where we can pass `parent_id` to get the IDs of the children!

Reply:
What about getting all folders once and then passing them around? I'd rather reduce the number of API calls.

Reply:
I think that's fine too. If it's a _huge_ tree of folders, it's possible that more Search calls could be faster than one Get All call, but I'm probably splitting hairs here.